worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java

```diff
@@ -133,40 +135,50 @@ private static boolean recreateShovel(
             final RetrievedShovel retrievedShovel,
             final ShovelsApi shovelsApi,
             final ShovelsApi nodeSpecificShovelsApi,
-            final String rabbitMQVHost
+            final String rabbitMQVHost,
+            final String rabbitMQUri
     )
     {
+        // - Shovels fetched from the /api/shovels/{vhost} endpoint (the RetrievedShovel parameter) do NOT include the
+        //   src-delete-after and ack-mode properties, which we need to recreate the shovel
+        // - Shovels fetched from the /api/parameters/shovel/{vhost} endpoint DO include the src-delete-after and ack-mode properties,
+        //   however, if a shovel gets stuck in a running state, the shovel may not be returned from the /api/parameters/shovel/{vhost}
+        //   endpoint, so if we can't get the shovel from /api/parameters/shovel/{vhost}, we do not know what the src-delete-after or
+        //   ack-mode properties should be
+        // - Therefore, we are recreating the shovel with a src-delete-after of 0, so that the shovel gets recreated then immediately
+        //   deleted. This allows the shovel to be created as normal (if still required) during the next loop of the application.
+
         final Shovel shovel = new Shovel();
-        shovel.setSrcDeleteAfter(retrievedShovel.getSrcDeleteAfter());
-        shovel.setAckMode(retrievedShovel.getAckMode());
+        shovel.setSrcDeleteAfter(0);
+        shovel.setAckMode(ACK_MODE);
```

Comment:
Given that some key parameters aren't available, we shouuld introduce a different data type to be returned from /api/shovels/{vhost}

Reply:
The existing class could extend this new class and add the additional fields.

Reply:
Any thoughts on what to rename these new classes to to make it clear what they should be used for?

Right now we have

- Shovel                                           (srcUri, destUri, srcQueue, destQueue, ackMode, srcDeleteAfter)
- RetrievedShovel extends Shovel   (ADDS: name, timestamp, state, node)


Should we change/rename these to:

- CommonShovelProps                           (srcUri, destUri, srcQueue, destQueue)
- GetShovelProps extends CommonShovelProps    (ADDS: name, timestamp, state, node)
- PutShovelProps extends CommonShovelProps    (ADDS: ackMode, srcDeleteAfter)

Reply:
I've gone with these classes for now, can rename these if required:

- CommonShovelProps
- RetrievedShovel extends CommonShovelProps
- ShovelToCreate extends CommonShovelProps

Commit: https://github.com/WorkerFramework/worker-message-prioritization/pull/36/commits/59ce89806224d5b4e92f1fe6e86724e32cc20c6a